### ISSUE TITLE
Initial allocation

### ISSUE THREAD

**Issue by gaurav1780** [gaurav1780/codingbat_python_project#1]
Hi all,

Thanks for volunteering! Remember, there is no rush to get it done soon. Even if we get it done by December, it's fine.
Also, feel free to raise doubts or ask questions here or email me.

STEP 1: Login

Username: gaurav1780@gmail.com
Password: python

STEP 2: Add questions at https://codingbat.com/author

# Screencast of sample porting to be shared soon, please don't start before that  since removing questions is not possible :(

# Tasks and allocation

* Variables @17JMW - https://codingbat.com/home/gaurav.gupta@mq.edu.au/variables to https://codingbat.com/home/gaurav1780@gmail.com/01_variables_and_operators
* Conditions @AiseLlamas - https://codingbat.com/home/gaurav.gupta@mq.edu.au/conditions to https://codingbat.com/home/gaurav1780@gmail.com/02_conditions
* Loops @alec-scappa - https://codingbat.com/home/gaurav.gupta@mq.edu.au/loops to https://codingbat.com/home/gaurav1780@gmail.com/03_loops
* Arrays @AtharvKamat1 - https://codingbat.com/home/gaurav.gupta@mq.edu.au/arrays to https://codingbat.com/home/gaurav1780@gmail.com/04_lists
* Recursion @emilywanczura - https://codingbat.com/home/gaurav.gupta@mq.edu.au/recursion to https://codingbat.com/home/gaurav1780@gmail.com/07_recursion
* Strings @Ty-swarris - https://codingbat.com/home/gaurav.gupta@mq.edu.au/strings to https://codingbat.com/home/gaurav1780@gmail.com/06_strings
* Bitwise @Mihir2077 - https://codingbat.com/home/gaurav.gupta@mq.edu.au/bitwise to https://codingbat.com/home/gaurav1780@gmail.com/08_bitwise
* Delegation @pixilised - https://codingbat.com/home/gaurav.gupta@mq.edu.au/strings to https://codingbat.com/home/gaurav1780@gmail.com/09_delegation

Thanks
Gaurav

**Comment by gaurav1780** [comment]
Ok, while I was recording the video for you all, I realized that adding test cases will be a giant pain in the backside unless they can be populated from my gaurav.gupta@mq.edu.au account. However, I cannot give out that password to everyone. 

So, the FIRST step has to be - one volunteer - who is willing to extract all test cases from my page, and populate them in an organized way (in an excel file or something) on github that others will use.

Please do not change any of the existing questions (or anything else on gaurav.gupta@mq.edu.au account on codingbat.com)

There are 161 problems, and assuming 2 minutes per problem, that is 5 hours of work. I can have a 10-minute meeting with whoever volunteers for this. 

Please reply as a comment if you are happy to help in this first step.

**Comment by ghost** [comment]
Hello,

I'm a bit confused about what the username for the account is meant to be.
The password, also; just for clarity, is it meant to be "# python" with a
space?

Regards,
Thomas Ling

**Comment by gaurav1780** [comment]
Hi Thomas and all,

Argh - my bad.

Username: gaurav1780@gmail.com
Password: python

@leviofliccat is porting all the tests into an excel file (sample attached). Once they are done, I will record a little screencast as a demo, and you can start writing the problems after that, copying and pasting the tests into the test box.

[codingbatTests.xlsx](https://github.com/gaurav1780/codingbat_python_project/files/12443627/codingbatTests.xlsx)

Cheers
Gaurav

**Comment by gaurav1780** [comment]
Can I check who has experience in python dictionaries so I can allocate that topic to them (as it's a new one - different from java)?

**Comment by AtharvKamat1** [comment]
I have experience in dictionaries,

**Comment by leviofliccat** [comment]
I also have some experience with dictionaries if you still need help after I finish the spreadsheet. I should hopefully get it done by this weekend :)

**Comment by leviofliccat** [comment]
Hi all, just letting you know that the test cases have been pushed to the GitHub repo. Let me know if there are any issues. Thanks

**Comment by gaurav1780** [comment]
Hi all, 

Thanks a tonne, Felicity! Fantastic work 🥇 Also, my wife (Gunjan) taught you COMP1000. She says 👋 

The allocation has been updated. Recording of the video is available at [https://macquarie.zoom.us/rec/share/mMFMetW6MbLKZE4buK6CjUmi2H2IvUoNFFoQNb4oPHtGVLFSWkuYHgE7h3JUte6j.986YogGt9_ofwudF](https://protect-au.mimecast.com/s/jExhC3QNl1Sk01Y7uDHrBR?domain=macquarie.zoom.us)

Once you watch the recording, you can get cracking.

Summary of key points:

- language:java -> lang:python
- showpost:true for all problems except for problems whose solutions you can see without solving it. for those, showsolution:true
- recursion depth is just under 1000 (for the problem i recorded). if depth is reached, remove the biggest test case and add a little smaller one, if possible.

IMPORTANT! Make sure you prioritize your own studies and well-being over this. My project can wait even after S2 finishes.
If you feel the work's too much or not for you, you can email me to discontinue at any stage.

Cheers, and Thanks,
Gaurav

**Comment by AtharvKamat1** [comment]
Hi,

When writing the problem description, would you like me to refer to them as arrays or lists? as codingbat has it labelled as lists.

Thanks, 
Atharv

**Comment by gaurav1780** [comment]
list, please!

**Comment by gaurav1780** [comment]
Hi all, lemme know if the video was sufficient or if you need any more help getting started.

**Comment by emilywanczura** [comment]
morning, the video was great thank you :))

**Comment by Maakusa** [comment]
So, do you want me to do anything?

**Comment by gaurav1780** [comment]
Hi @Maakusa, SINCERE apologies. I realize you haven't been allocated anything. My bad :(((
Can I please request you to take over LOOPS?

So far it's recursion that has been completed in its entirety (thanks Emily) and lists which is almost there (thanks @AtharvKamat1).

For the others, please let me know if you are still interested, otherwise I can allocate your part to someone else, or do it myself.

@leviofliccat would you like to take over variables OR conditions?

Cheers,
Gaurav

**Comment by leviofliccat** [comment]
Sure, I can have a crack at conditions :)

**Comment by gaurav1780** [comment]
All right then! Conditions transferred from @AiseLlamas to @leviofliccat. Please do prioritize your studies. It's absolutely fine to do this after the session finishes :) And thanks again 🙏

**Comment by sarthdan** [comment]
@gaurav1780 I am happy to take over variables if required.

**Comment by gaurav1780** [comment]
@sarthdan yes please. But only when you have spare time. Your studies come first.

**Comment by AtharvKamat1** [comment]
Hi all,

I have just about finished the lists/arrays section; however, I have had some issues with a few problems and haven't been able to figure them out in python. If anyone has some spare time and are able to do these problems, please help me out and upload them on codingbat if possible.

Level 3: IsSubsetOf (p212371)
Level 4: frequencyTable (p226684)
Level 5: oneInEachSubArray (p273221)

Cheers,
Atharv

**Comment by leviofliccat** [comment]
Hey Atharv,

I've completed and uploaded those three questions to codingbat. :) Let me know if there are any problems.

Cheers,
Felicity

**Comment by Mihir2077** [comment]
Hi all,

I have finished the 08_bitwise section; sorry for the delay. I am happy to contribute to the strings sections.

Thanks,
Mihir

**Comment by gaurav1780** [comment]
Hi Mihir

That's awesome. Yeah, if you have time, please add Strings questions.

Cheers
Gaurav

**Comment by Mihir2077** [comment]
Hi all, 

I have finished the 06_strings section except for the following two questions. I am finding it difficult to do the problems. 

Level Oogway
- divide  -->  p238007
- remainder --> p210171

Cheers,
Mihir

**Comment by gaurav1780** [comment]
All good, Mihir. Tha is a tonne for doing the rest :)

G

**Comment by Maakusa** [comment]
Hi all,

Just finished what was left of the 03_loops section. Sorry for doing it so late

Cheers,
Mark

**Comment by gaurav1780** [comment]
Awesome, thanks for the update!
